### PR TITLE
Added board for ESP32-S2-WROVER module

### PR DESCRIPTION
**Pull request by ThinkTransit**
Hi,

This PR is to add a board for the Espressif ESP32-S2-WROVER module.

4 MB SPI flash
2 MB PSRAM

### CODE REVIEW

File: ports/esp32/boards/ESP32_S2_WROVER/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#define MICROPY_HW_BOARD_NAME "ESP32-S2-WROVER"
+#define MICROPY_HW_MCU_NAME "ESP32-S2"
+
+#define MICROPY_PY_BLUETOOTH                (0)
+#define MICROPY_HW_ENABLE_SDCARD            (0)
+
+#define MICROPY_HW_I2C0_SCL (9)
+#define MICROPY_HW_I2C0_SDA (8)
```
Comment (dpgeorge) [inline ports/esp32/boards/ESP32_S2_WROVER/mpconfigboard.h:8]: I2C1 has default pins 9 and 8, so with this configuration I2C(0) and I2C(1) will use the same pins.  Does it make sense to configure them differently (eg put I2C(0) on other pins)?

File: ports/esp32/boards/ESP32_S2_WROVER/mpconfigboard.h [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+#define MICROPY_HW_SPI1_MOSI (35)  // SDO
+#define MICROPY_HW_SPI1_MISO (37)  // SDI
+#define MICROPY_HW_SPI1_SCK (36)
```
Comment (dpgeorge) [inline ports/esp32/boards/ESP32_S2_WROVER/mpconfigboard.h:12]: There is also SPI(2), with defaults of pins 18/19/23.  Is that correct for this board?

File: ports/esp32/boards/ESP32_S2_WROVER/mpconfigboard.h (hunk 1 above)
Comment (ThinkTransit) [reply ports/esp32/boards/ESP32_S2_WROVER/mpconfigboard.h]: Good point I have moved I2C(0).

File: ports/esp32/boards/ESP32_S2_WROVER/mpconfigboard.h (hunk 2 above)
Comment (ThinkTransit) [reply ports/esp32/boards/ESP32_S2_WROVER/mpconfigboard.h]: No 18/19/23 is not correct for the ESP32-S2, I have updated the pins for both SPI's according the the module datasheet.

**Comment by ThinkTransit** [comment]
Ok this is ready to go now, I was getting the chip pin numbers and the GPIO numbers mixed up!

SPI

**Comment by dpgeorge** [comment]
Merged in 4cfd85eb4a4dbab02efb1b22e87789f552e68fc0 with minor formatting updates, and also removed the board-specific `manifest.py` and used the generic one in `boards/manifest.py`, because all it did was include the generic one.

Thanks for the contribution!

**Pull request closed without merging**